pkg/controller/endpointslice/utils.go

```diff
@@ -87,6 +87,10 @@ func podToEndpoint(pod *corev1.Pod, node *corev1.Node, service *corev1.Service,
 		ep.Conditions.Terminating = &terminating
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.EndpointSliceNodeName) {
+		ep.NodeName = &pod.Spec.NodeName
```

Comment:
If `pod.Spec.NodeName` is empty, should we set `ep.NodeName` to nil instead?

Reply:
Updated, thanks!